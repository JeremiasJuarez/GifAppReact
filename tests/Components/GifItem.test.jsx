import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/Components/GifItem"

describe('Pruebas en <GifItem/>', () => { 

    const url = 'https://www.link.com/snapacshotasdafdkm123213'
    const title = 'Titulo'

    test('Debe recibir title y url obligatoriamente', () => { 

        render( <GifItem url={ url } title={ title }/> )

     })

     test('Debe hacer match con el snapshot', () => { 

        const { container } = render(<GifItem url={url} title={title}/>)

        expect( container ).toMatchSnapshot()

      })

      test('Debe mostrar URL y ALT correctos', () => { 

        render(<GifItem url={url} title={title}/>)
        
        // expect(screen.getByRole('img').src).toBe( url )

        const { src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)

      })

      test('Debe mostrar el titulo en el componente', () => { 

        render(<GifItem url={url} title={title}/>)
        expect( screen.getByText( title )).toBeTruthy()

       })
 })