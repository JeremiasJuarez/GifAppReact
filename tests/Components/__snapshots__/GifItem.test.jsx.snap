// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo"
      src="https://www.link.com/snapacshotasdafdkm123213"
    />
    <p>
      Titulo
    </p>
  </div>
</div>
`;
